### ISSUE TITLE
Strange cursor color on some lines

### ISSUE THREAD

**Issue by FelixKratz** [alacritty/alacritty#5440]
### System

OS: macOS
Version: alacritty 0.9.0 (fed349a)

Without any configuration file

#########
I experience inconsistent cursor colors, where the cursor on an empty line sometimes looks like this:
![Screen Shot 2021-08-22 at 16 31 52](https://user-images.githubusercontent.com/22680421/130359173-2d29e78c-45fa-4c4c-835a-8a90034ce361.png)
and sometimes like this (which I would expect on all lines):
![Screen Shot 2021-08-22 at 16 31 35](https://user-images.githubusercontent.com/22680421/130359178-7d34f4f1-6285-4a20-af36-dccb403930ba.png)

**Comment by chrisduerr** [comment]
Likely duplicate of #4564.

**Comment by FelixKratz** [comment]
Any way to deactivate the contrast logic? Seems to ruin the experience for me.

**Comment by chrisduerr** [comment]
Not without patching right now unfortunately.

**Comment by FelixKratz** [comment]
I have compiled the app with
```rust
pub const MIN_CURSOR_CONTRAST: f64 = 0.0;
```
and experience the exact same behaviour.

**Comment by FelixKratz** [comment]
I have forced alacritty to always display the foreground color as the background color:
![Screen Shot 2021-09-03 at 10 45 00](https://user-images.githubusercontent.com/22680421/131977653-06c6577b-3b95-4e9b-b63b-2ddbda0a8fa2.png)
it is obvious that the contrast logic is not the culprit here. Rather, the foreground color of an empy cell is assumed to be blue. It would be great if I could set a default cursor color that is used when no actual foreground color is present, would such thing be possible?

**Comment by chrisduerr** [comment]
> It would be great if I could set a default cursor color that is used when no actual foreground color is present

"no actual foreground color is present" doesn't exist. Every cell has a foreground color and it's up to applications to set them.

**Comment by FelixKratz** [comment]
Ah, got it, thanks! Then the problem was with my nvim all along and not with alacritty. It is curious though, because in iTerm, for example, the cursor stays in the "correct" color in these spots.

**Comment by kirusfg** [comment]
Hi @FelixKratz, is that NvChad you are using, or at least the `indent-blankline` plugin for vim? People, including me, have been having this issue in `alacritty` and `xterm`. Were you able to resolve it? 

I have browsed this repository for similar issues, and it is apparently not the issue within `alacritty` itself, but rather in the `vim` configuration. Then I experimented a bit and discovered that when opening the file using `nvim` command in the terminal, like so:
```
nvim file.c
```
the blue cursor issue happens. Opening the file through a file manager like `ranger` is the same, because the file is opened in the same way.

But if the file is opened within the `nvim` itself (through telescope or fzf, for example), the issue is not there. This could mean that a plugin or the plugin's settings do not get loaded up in time for the file to be displayed correctly.

**TL;DR** The issue is not with `alacritty`, but with `vim` configuration. I would like to reduce the scope of the issue down to a single plugin or the way settings are loaded on `vim` startup.

**Comment by FelixKratz** [comment]
Hi @kirusfg, I found the culprit to be the `M.reset_highlights` function in the indent-blankline plugin. It resets for ctermfg color here:
```lua
....
            else
                vim.cmd(
                    string.format(
                        "highlight %s guifg=%s ctermfg=%s gui=nocombine cterm=nocombine",
                        highlight_name,
                        M._if(highlight[1] == "", "NONE", highlight[1]),
                        M._if(highlight[2] == "", "NONE", highlight[2])
                    )
                )
            end
``` 
if you simply remove the whole vim.cmd the issue is resolved.
